src/vulkan-renderer/wrapper/cpu_texture.cpp

```diff
@@ -0,0 +1,87 @@
+
+    if (texture_data == nullptr) {
+        spdlog::error("Could not load texture file {} using stbi_load!  Falling back to error texture.", file_name);
+        generate_error_texture_data();
+    } else {
+
+        // TODO: We are currently hard coding the number of channels with STBI_rgb_alpha.
+        //       Eventually, we probably need to pass this information into this class from
+        //       a higher level class - like a material loader class.
+        //       So, as an example, if the material loader is loading a normal map, we know
+        //       we need to tell this class to load a 3 channel texture.  If it can, great.
+        //       If it can not, then this class probably needs to load a 3 channel error texture.
+        m_texture_channels = 4;
+
+        // TODO: We are currently only supporting 1 mip level.
+        m_mip_levels = 1;
+
+        const int texture_data_size{m_texture_width * m_texture_height * m_texture_channels};
+        m_texture_data.resize(texture_data_size);
+        std::copy(texture_data, texture_data + texture_data_size, m_texture_data.begin());
+
+        spdlog::debug("Texture dimensions: width: {}, height: {}, channels: {} mip levels: {}.", m_texture_width,
+                      m_texture_height, m_texture_channels, m_mip_levels);
+    }
+
+    // Discard the texture data.
+    stbi_image_free(texture_data);
+}
+
+CpuTexture::CpuTexture(CpuTexture &&other) noexcept
+    : m_file_name(std::move(other.m_file_name)), m_name(std::move(other.m_name)),
+      m_texture_width(other.m_texture_width), m_texture_height(other.m_texture_height),
+      m_texture_channels(other.m_texture_channels), m_mip_levels(other.m_mip_levels),
+      m_texture_data(std::move(other.m_texture_data)) {}
+
+CpuTexture::~CpuTexture() {}
+
+void CpuTexture::generate_error_texture_data() {
+    m_texture_width = 512;
+    m_texture_height = 512;
+    m_texture_channels = 4;
+    m_mip_levels = 1;
+
+    // Create an 8x8 checkerboard pattern of squares.
+    const std::size_t square_dimension{64};
+    // pink, purple
+    constexpr std::array<std::array<unsigned char, 4>, 2> COLORS{{{0xFF, 0x69, 0xB4, 0xFF}, {0x94, 0x00, 0xD3, 0xFF}}};
+
+    const auto get_color = [](std::size_t x, std::size_t y, std::size_t square_dimension,
+                              std::size_t colors) -> std::size_t {
+        return (std::size_t(x / square_dimension) + std::size_t(y / square_dimension)) % colors;
+    };
+
+    m_texture_data.reserve(m_texture_width * m_texture_height * m_texture_channels);
+    for (std::size_t y = 0; y < m_texture_height; y++) {
+        for (std::size_t x = 0; x < m_texture_width; x++) {
+            const std::size_t color_id = get_color(x, y, square_dimension, COLORS.size());
+            m_texture_data.insert(m_texture_data.end(), std::begin(COLORS[color_id]), std::end(COLORS[color_id]));
+        }
+    }
```

Comment:
Use `int` instead of `std::size_t`, as the width and height is defined as int.

Reply:
Fixed.